src/native/corehost/apphost/apphost.windows.cpp

```diff
@@ -86,7 +89,52 @@ namespace
         return msg;
     }
 
-    void show_error_dialog(const pal::char_t *executable_name, int error_code)
+    class activation_context final
+    {
+    public:
+        activation_context(const pal::char_t* source, const pal::char_t* resource_name)
+            : m_activated{false}
+            , m_context_handle{INVALID_HANDLE_VALUE}
+            , m_cookie{}
+        {
+            ACTCTXW actctx = { sizeof(ACTCTXW), 0, source };
+            if (resource_name != nullptr)
+            {
+                actctx.lpResourceName = resource_name;
+                actctx.dwFlags = ACTCTX_FLAG_RESOURCE_NAME_VALID;
+            }
+
+            m_context_handle = ::CreateActCtxW(&actctx);
+            if (m_context_handle == INVALID_HANDLE_VALUE)
+                return;
+
+            if (::ActivateActCtx(m_context_handle, &m_cookie) == FALSE)
+                return;
+
+            m_activated = true;
+        }
+
+        ~activation_context()
+        {
+            if (m_activated)
+                ::DeactivateActCtx(0, m_cookie);
+
+            if (m_context_handle != INVALID_HANDLE_VALUE)
+                ::ReleaseActCtx(m_context_handle);
+        }
+
+        bool activated()
```

Comment:
Doesn't seem to be used by anything.
(I like the robustness of the class, but since this is only used here we might as well make it do only the things we need and no more)